lib/generators/deimos/schema_class_generator.rb

```diff
@@ -0,0 +1,231 @@
+        # @return [String]
+        def field_to_h(field)
+          res = "'#{field.name}' => @#{field.name}"
+          field_base_type = _schema_base_type(field.type).type_sym
+
+          if %i(record enum).include?(field_base_type)
+            res += case field.type.type_sym
+                   when :array
+                     '.map { |v| v&.to_h }'
+                   when :map
+                     '.transform_values { |v| v&.to_h }'
+                   else
+                     '&.to_h'
+                   end
+          end
+
+          res + (field.name == fields.last.name ? '' : ',')
+        end
+
+      end
+
+      desc 'Generate a class based on an existing schema.'
+      # :nodoc:
+      def generate
+        Rails.logger.info(Deimos.config.schema.path)
+        if full_schema.nil?
+          _find_schema_paths.each do |schema_path|
+            current_schema = _parse_schema_from_path(schema_path)
+
+            generate_classes_from_schema(current_schema)
+          end
+        else
+          current_schema = _parse_schema_from_path(full_schema)
+          generate_classes_from_schema(current_schema)
+        end
+      end
+
+    private
+
+      # Retrieve all Avro Schemas under the configured Schema path
+      # @return [Array<String>] array of the full path to each schema in schema.path.
+      def _find_schema_paths
+        Dir["#{_schema_path}/**/*.avsc"]
+      end
+
+      def _schema_path
+        Deimos.config.schema.path || File.expand_path('app/schemas', __dir__)
+      end
+
+      # Load the schema from the Schema Backend
+      # @param schema [String] the Schemas name
+      # @return [Deimos::SchemaBackends::Base]
+      def _schema_base(schema=nil)
+        @schema_base ||= Deimos.schema_backend_class.new(schema: extract_schema(schema),
+                                                         namespace: extract_namespace(schema))
+      end
+
+      # Unload the schema
+      def _clear_schema_base!
+        @schema_base = nil
+      end
+
+      # Parses the schema in dot syntax from a given Schema Path.
+      # Handles different cases for File/Schema names.
+      # @return [String] The name of the schema in the format of com.my.namespace.MySchema
+      def _parse_schema_from_path(schema_path)
+        return schema_path unless schema_path =~ %r{/}
+
+        full_schema_path = File.absolute_path(schema_path)
+        schema_name = File.basename(full_schema_path, '.avsc')
+        namespace_dir = File.dirname(full_schema_path).
+          delete_prefix("#{_schema_path}/").
+          gsub('/', '.')
+
+        "#{namespace_dir}.#{schema_name}"
+      end
+
+      # Converts Avro::Schema::NamedSchema's to String form for generated YARD docs.
+      # Recursively handles the typing for Arrays, Maps and Unions.
+      # @param avro_schema [Avro::Schema::NamedSchema]
+      # @return [String] A string representation of the Type of this SchemaField
+      def _field_type(avro_schema)
```

Comment:
E.g. this method could go into avro_base.

Reply:
noice - i like this method! lets promote it